### ISSUE TITLE
WARNING:root:The model is invalid: Unable to open proto file:

### ISSUE THREAD

**Issue by banzhaoyang** [vietanhdev/anylabeling#113]
WARNING:root:The model is invalid: Unable to open proto file: C:\Users\admin、\anylabeling_data\models\segment_anything_vit_b_quant-r20230416\segment_anything_vit_b_encoder_quant.onnx. Please check if it is a valid proto.
WARNING:root:Action: Delete and redownload...
Could not download https://github.com/vietanhdev/anylabeling-assets/releases/download/v0.2.0/segment_anything_vit_b_encoder_quant.onnx: <urlopen error retrieval incomplete: got only 98777600 out of 99827815 bytes>
Error in loading model: Could not download or initialize encoder of Segment Anything.

**Comment by banzhaoyang** [comment]
这是什么情况 重新下载了onnx模型，也用不了

**Comment by vietanhdev** [comment]
Please check your internet connection and download the model again. Thank you!

**Comment by vietanhdev** [comment]
The original failure was an interrupted network transfer (`retrieval incomplete`), not an invalid SAM graph. The current model catalog no longer uses that retired individual-file download: SAM variants are distributed as versioned archives, downloaded into a temporary directory, and installed only after the archive is complete and contains `config.yaml`. A failed transfer leaves `has_downloaded: false`, so selecting the model again retries instead of treating a partial file as installed.

Current `main` also catches download/archive/model-initialization errors inside the worker, reports the error in the UI, emits completion, and returns the model picker to an idle state rather than leaving a stranded thread (#262).

If a current release still cannot download a model, please open a new issue with the selected model name, AnyLabeling version, and complete error; that would be a new endpoint/network-specific failure rather than this retired v0.2 asset.
